### ISSUE TITLE
How loss calculation is implemented in Faster RCNN

### ISSUE THREAD

**Issue by nyanmn** [rbgirshick/py-faster-rcnn#756]
I roughly understand the loss calculation in Faster RCNN.
Every layer calculates the loss and do summation at the end of one iteration.

But my query is how Loss function in equation (1) of this paper ([https://arxiv.org/pdf/1506.01497.pdf](url)) is implemented in Loss calculation.

I run through the code and found the implementation of  SoftmaxWithLoss and SmoothL1LossLayer.

But I can't find how this predicted and ground truth boxes are used in loss calculation? Where they are implemented?

Can somebody help me?
